NumPy.py

```diff
@@ -1,12 +1,236 @@
 import numpy as np
+import matplotlib.pyplot as plt
 
 
 # ----------q1----------
-def change_sign():
-    array = np.array(range(21))
-    np.copysign(array[9:16], -1)
-    return array
+def question1():
+    """
+    The function creates a vector with values from 0 to 20 and change the sign of the numbers in the range from 9 to 15.
+    :return: The vector after the change.
+    """
+    vector = np.array(range(21))
+    vector[9:16] *= -1
+    return vector
+
+
+# ----------q2----------
+def question2():
+    """
+    The function creates a vector of length 10 with values evenly distributed between 5 and 50.
+    :return: The vector that created.
+    """
+    return np.linspace(5, 50, 10)
+
+
+# ----------q3----------
+def question3(matrix):
+    """
+    The function finds number of rows and columns of a given matrix.
+    :param matrix: The given matrix.
+    :return: The matrix after the change.
+    """
+    return np.array(matrix).shape
+
+
+# ----------q4----------
+def question4():
+    """
+    The function creates a 10x10 matrix, in which the elements on the borders will be equal to 1, and inside 0.
+    :return: The matrix that created.
+    """
+    matrix = np.ones((10, 10))
+    matrix[1:-1, 1:-1] = 0
+    return matrix
+
+
+# ----------q5----------
+def question5(matrix, vector):
+    """
+    The function adds the vector to each row of a given matrix.
+    :param matrix: The given matrix.
+    :param vector: The vector that adds.
+    :return: The matrix after change.
+    """
+    matrix_after_add = np.empty_like(matrix)
+    for row in range(len(matrix)):
+        matrix_after_add[row, :] = matrix[row, :] + vector
```

Comment:
The beauty of numpy is that in most cases for loops are not needed (and are usually less efficient), try using broadcasting.
https://numpy.org/doc/stable/user/basics.broadcasting.html